File: Year-2023/LeetCode-January-Challenge/FlipStringtoMonotoneIncreasing.java
class Solution {
    public int minFlipsMonoIncr(String s) {
        int minFlips = 0, counter = 0;
        for(char ch: s.toCharArray()){
            if(ch == '1') counter++;
            else minFlips++;
            minFlips = Math.min(minFlips, counter);
        }
        return minFlips;
    }
}
public class FlipStringtoMonotoneIncreasing{
    public static void main(String[] args) {
        String s = "00110";
        Solution S = new Solution();
        System.out.println(S.minFlipsMonoIncr(s));
    }
}

File: Year-2023/LeetCode-January-Challenge/FlipStringtoMonotoneIncreasing.js
var minFlipsMonoIncr = function(s) {
    let minFlips = 0, counter = 0;
    for(let ch of s){
        if(ch === '1') counter++;
        else minFlips++;
        minFlips = Math.min(minFlips, counter);
    }
    return minFlips;
};
let s = "00110";
console.log(minFlipsMonoIncr(s));